astropy/modeling/tests/test_input.py

```diff
@@ -205,11 +205,11 @@ def test_nonlinear_lsqt_Nset_1d(self, fitter):
         fitter = fitter()
 
         MESSAGE = r"Non-linear fitters can only fit one data set at a time"
+        g1 = models.Gaussian1D(
+            [10.2, 10], mean=[3, 3.2], stddev=[0.23, 0.2], n_models=2
+        )
+        y1 = g1(self.x1, model_set_axis=False)
         with pytest.raises(ValueError, match=MESSAGE):
-            g1 = models.Gaussian1D(
-                [10.2, 10], mean=[3, 3.2], stddev=[0.23, 0.2], n_models=2
-            )
-            y1 = g1(self.x1, model_set_axis=False)
             _ = fitter(g1, self.x1, y1)
```

Comment:
```suggestion
            fitter(g1, self.x1, y1)
```
There is no use in even invoking the assignment here

Reply:
Without the assignment, wouldn't this be removed by https://docs.astral.sh/ruff/rules/useless-expression/?

Reply:
It does not trigger the issue when I run ruff locally with these changes.